src/main/java/de/tum/in/www1/artemis/repository/ResultRepository.java

```diff
@@ -467,4 +483,7 @@ default DueDateStat countNumberOfAssessmentsOfExam(Long courseId) {
             """)
     List<TutorLeaderboardAssessments> findTutorLeaderboardAssessmentByExamId(@Param("examId") long examId);
 
+    default Set<Long> mapExerciseToId(Set<Exercise> exercises) {
+        return exercises.stream().map(exercise -> exercise.getId()).collect(Collectors.toSet());
```

Comment:
I suppose you can use `Exercise::getId` instead of `exercise -> exercise.getId()` here, but that is not important.

Reply:
You are right, that's more concise.